### PR TITLE
Implement `Send` for chunk types

### PR DESCRIPTION
**Pull request by RamType0**
#4

### CODE REVIEW

**Comment by RamType0** [comment]
This clippy error is really weird
>error: this implementation is unsound, as some fields in `Producer<T>` are `!Send`
   --> src/lib.rs:291:1
    |
291 | unsafe impl<T: Send> Send for Producer<T> {}
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: `-D clippy::non-send-fields-in-send-ty` implied by `-D warnings`
note: the type of field `buffer` is `!Send`
   --> src/lib.rs:278:5
    |
278 |     buffer: Arc<RingBuffer<T>>,
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^
    = help: add bounds on type parameter `T` that satisfy `Arc<RingBuffer<T>>: Send`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#non_send_fields_in_send_ty

>error: this implementation is unsound, as some fields in `Consumer<T>` are `!Send`
   --> src/lib.rs:496:1
    |
496 | unsafe impl<T: Send> Send for Consumer<T> {}
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
note: the type of field `buffer` is `!Send`
   --> src/lib.rs:483:5
    |
483 |     buffer: Arc<RingBuffer<T>>,
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^
    = help: add bounds on type parameter `T` that satisfy `Arc<RingBuffer<T>>: Send`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#non_send_fields_in_send_ty

>error: could not compile `rtrb` due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
error: build failed
Error: Process completed with exit code 101.

https://github.com/rust-lang/rust-clippy/issues/8045

**Comment by mgeier** [comment]
Yeah, that's strange.
AFAIU, this lint will be disabled in the next point release which comes out within the next few days.

Interestingly, this lint is not triggered by #75.

**Comment by mgeier** [comment]
Rust 1.58.1 has been release, I've restarted CI and now it passes.

File: src/chunks.rs [hunk 1]
```diff
@@ -444,6 +444,8 @@ pub struct WriteChunkUninit<'a, T> {
     producer: &'a Producer<T>,
 }
 
+unsafe impl<T: Send> Send for WriteChunkUninit<'_, T> {}
```
Comment (mgeier) [inline src/chunks.rs:447]: What about adding a comment regarding safety?

Maybe something like this:

```
// WriteChunkUninit only exists while a unique reference to the Producer is held.
// It is therefore safe to move it to another thread.
```

File: src/chunks.rs [hunk 2]
```diff
@@ -614,6 +616,8 @@ pub struct ReadChunk<'a, T> {
     consumer: &'a mut Consumer<T>,
 }
 
+unsafe impl<T: Send> Send for ReadChunk<'_, T> {}
```
Comment (mgeier) [inline src/chunks.rs:619]:
```suggestion
// ReadChunk only exists while a unique reference to the Consumer is held.
// It is therefore safe to move it to another thread.
unsafe impl<T: Send> Send for ReadChunk<'_, T> {}
```

**Review by mgeier — COMMENTED** [review]
I've added comments as suggestions, what do you think about them?

**Pull request merged**